### PR TITLE
Feat/#347 ErrGasUintOverflow constraint

### LINKED ISSUE #347: missing ErrGasUintOverflow error state
need to add it into https://github.com/privacy-scaling-explorations/zkevm-specs/blob/master/src/zkevm_specs/evm/execution_state.py#L81

### PR DESCRIPTION
**Pull request by KimiWu123**
This is updated version of #361 which was trying to fix #347.

#361 has been approved by @hero78119 and @KimiWu123. What I'm doing here is to finish the test and major changes are under [eecc9a2](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/436/commits/eecc9a271f6aded3d4a477c3997900657d64ef9c)

### CODE REVIEW

File: src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                )
+
+            # TODO: Would like to support EIP 3860 in the future (See
+            # https://github.com/privacy-scaling-explorations/zkevm-specs/issues/421)
+            # gas += z * TxDataZeroGas
+            # if is_create:
+            #     lenWords = data_len // 32
+            #     is_eip3860_overflow, _ = instruction.compare(
+            #         FQ((MAX_U64 - gas) // InitCodeWordGas), FQ(lenWords), N_BYTES_U64
+            #     )
```
Comment (ChihChengLiang) [inline src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py:136]: Are these leftover or part of the TODO comments?

**Review by ChihChengLiang — APPROVED** [review]
LGTM.
I only did a quick look as you said this PR is already approved in https://github.com/privacy-scaling-explorations/zkevm-specs/pull/361 and the goal of this PR is to fix tests.
Great work fixing tests.
I added a nitpick.

File: src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py (hunk 1 above)
Comment (KimiWu123) [reply src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py:136]: It looks like a part of TODO comments. Do you think we need to remove it?

**Comment by KimiWu123** [comment]
Since https://github.com/privacy-scaling-explorations/zkevm-specs/pull/361 has been approved, I think one approve is enough for this PR. How do you think, @ChihChengLiang ?

File: src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py (hunk 1 above)
Comment (ChihChengLiang) [reply src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py:136]: I think we can remove them so that people won't get confused.

**Comment by ChihChengLiang** [comment]
I think one approval is okay. Let me know if you can force merge it after your final patch.

**Pull request merged**